backend/app/parsers.py

```diff
@@ -1,39 +1,70 @@
+from functools import partial
+
 from flask_restful import reqparse
-from sqlalchemy import exists
 
-from app.models import Category, Subcategory
+from app.models import Category, Subcategory, Item, User
 from init_app import db
 
 
-def check_category(value):
-    if not db.session.query(exists().where(Category.name == value)).scalar():
-        raise ValueError("Value is not Category instance")
+def object_exist(model, value, attribute):
+    object_attribute = getattr(model, attribute)
+    return db.session.query(model.query.filter(object_attribute == value).exists()).scalar()
+
+
+def check_object(model, value, attribute='name'):
+    if not object_exist(model, value, attribute):
+        raise ValueError(f"{value} is not class instance")
     return value
 
 
-def check_subcategory(value):
-    if not db.session.query(exists().where(Subcategory.name == value)).scalar():
-        raise ValueError("Value is not Subcategory instance")
+def check_unique_object(model, value, attribute='name'):
```

Comment:
I know I suggested making partial function from this function, it was excercise for you. Is this function really needed? Do you make unique constraint on the models?

Reply:
Yes, I do